### PR TITLE
Integrate basic services

### PR DESCRIPTION
**Pull request by at00825957**
Integrate basic services #80

### CODE REVIEW

File: server/controllers/BasicServices.js [hunk 1]
```diff
@@ -1,164 +1,320 @@
...
 
-module.exports.redirectOamRequestInformation = function redirectOamRequestInformation (req, res, next, body, user, originator, xCorrelator, traceIndicator, customerJourney) {
-  BasicServices.redirectOamRequestInformation(body, user, originator, xCorrelator, traceIndicator, customerJourney)
-    .then(function (response) {
-      utils.writeJson(res, response);
+module.exports.redirectOamRequestInformation = async function redirectOamRequestInformation (req, res, next, body, user, originator, xCorrelator, traceIndicator, customerJourney) {
+  let startTime = process.hrtime();
+  let responseCode = responseCodeEnum.code.NO_CONTENT;
+  let responseBodyToDocument = {};
+  await BasicServices.redirectOamRequestInformation(body, user, xCorrelator, traceIndicator, customerJourney, req.url, NEW_RELEASE_FORWARDING_NAME)
```
Comment (PrathibaJee) [inline server/controllers/BasicServices.js:166]: undefined variable "**NEW_RELEASE_FORWARDING_NAME**" causing the following error and terminating the application , 

```

ReferenceError: NEW_RELEASE_FORWARDING_NAME is not defined
    at SwaggerRouter.redirectServiceRequestInformation (..\AirInterfacePowerSaver\server\controllers\BasicServices.js:185:124)
    at ..\AirInterfacePowerSaver\server\node_modules\openbackhaul-oas3-tools\dist\middleware\swagger.router.js:99:40
    at Layer.handle [as handle_request] (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\layer.js:95:5)
    at trim_prefix (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\index.js:328:13)
    at ..\AirInterfacePowerSaver\server\node_modules\express\lib\router\index.js:286:9
    at Function.process_params (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\index.js:346:12)
    at next (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\index.js:280:10)
    at ..\AirInterfacePowerSaver\server\node_modules\openbackhaul-oas3-tools\dist\middleware\swagger.parameters.js:52:13
    at Layer.handle [as handle_request] (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\layer.js:95:5)
    at trim_prefix (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\index.js:328:13)
```

File: server/controllers/BasicServices.js [hunk 2]
```diff
@@ -1,164 +1,320 @@
...
 
-module.exports.redirectServiceRequestInformation = function redirectServiceRequestInformation (req, res, next, body, user, originator, xCorrelator, traceIndicator, customerJourney) {
-  BasicServices.redirectServiceRequestInformation(body, user, originator, xCorrelator, traceIndicator, customerJourney)
-    .then(function (response) {
-      utils.writeJson(res, response);
+module.exports.redirectServiceRequestInformation = async function redirectServiceRequestInformation (req, res, next, body, user, originator, xCorrelator, traceIndicator, customerJourney) {
+  let startTime = process.hrtime();
+  let responseCode = responseCodeEnum.code.NO_CONTENT;
+  let responseBodyToDocument = {};
+  await BasicServices.redirectServiceRequestInformation(body, user, xCorrelator, traceIndicator, customerJourney, req.url, NEW_RELEASE_FORWARDING_NAME)
```
Comment (PrathibaJee) [inline server/controllers/BasicServices.js:185]: undefined variable "**NEW_RELEASE_FORWARDING_NAME**" causing the following error and terminating the application , 

```

ReferenceError: NEW_RELEASE_FORWARDING_NAME is not defined
    at SwaggerRouter.redirectServiceRequestInformation (..\AirInterfacePowerSaver\server\controllers\BasicServices.js:185:124)
    at ..\AirInterfacePowerSaver\server\node_modules\openbackhaul-oas3-tools\dist\middleware\swagger.router.js:99:40
    at Layer.handle [as handle_request] (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\layer.js:95:5)
    at trim_prefix (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\index.js:328:13)
    at ..\AirInterfacePowerSaver\server\node_modules\express\lib\router\index.js:286:9
    at Function.process_params (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\index.js:346:12)
    at next (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\index.js:280:10)
    at ..\AirInterfacePowerSaver\server\node_modules\openbackhaul-oas3-tools\dist\middleware\swagger.parameters.js:52:13
    at Layer.handle [as handle_request] (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\layer.js:95:5)
    at trim_prefix (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\index.js:328:13)
```

File: server/controllers/BasicServices.js [hunk 3]
```diff
@@ -1,164 +1,320 @@
...
 
-module.exports.redirectTopologyChangeInformation = function redirectTopologyChangeInformation (req, res, next, body, user, originator, xCorrelator, traceIndicator, customerJourney) {
-  BasicServices.redirectTopologyChangeInformation(body, user, originator, xCorrelator, traceIndicator, customerJourney)
-    .then(function (response) {
-      utils.writeJson(res, response);
+module.exports.redirectTopologyChangeInformation = async function redirectTopologyChangeInformation (req, res, next, body, user, originator, xCorrelator, traceIndicator, customerJourney) {
+  let startTime = process.hrtime();
+  let responseCode = responseCodeEnum.code.OK;
+  let responseBodyToDocument = {};
+  await BasicServices.redirectTopologyChangeInformation(body, user, xCorrelator, traceIndicator, customerJourney, req.url, NEW_RELEASE_FORWARDING_NAME)
```
Comment (PrathibaJee) [inline server/controllers/BasicServices.js:204]: undefined variable "**NEW_RELEASE_FORWARDING_NAME**" causing the following error and terminating the application , 

```

ReferenceError: NEW_RELEASE_FORWARDING_NAME is not defined
    at SwaggerRouter.redirectServiceRequestInformation (..\AirInterfacePowerSaver\server\controllers\BasicServices.js:185:124)
    at ..\AirInterfacePowerSaver\server\node_modules\openbackhaul-oas3-tools\dist\middleware\swagger.router.js:99:40
    at Layer.handle [as handle_request] (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\layer.js:95:5)
    at trim_prefix (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\index.js:328:13)
    at ..\AirInterfacePowerSaver\server\node_modules\express\lib\router\index.js:286:9
    at Function.process_params (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\index.js:346:12)
    at next (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\index.js:280:10)
    at ..\AirInterfacePowerSaver\server\node_modules\openbackhaul-oas3-tools\dist\middleware\swagger.parameters.js:52:13
    at Layer.handle [as handle_request] (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\layer.js:95:5)
    at trim_prefix (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\index.js:328:13)
```

File: server/controllers/BasicServices.js [hunk 4]
```diff
@@ -1,164 +1,320 @@
...
 
-module.exports.updateClient = function updateClient (req, res, next, body, user, originator, xCorrelator, traceIndicator, customerJourney) {
-  BasicServices.updateClient(body, user, originator, xCorrelator, traceIndicator, customerJourney)
-    .then(function (response) {
-      utils.writeJson(res, response);
+module.exports.updateClient = async function updateClient (req, res, next, body, user, originator, xCorrelator, traceIndicator, customerJourney) {
+  let startTime = process.hrtime();
+  let responseCode = responseCodeEnum.code.NO_CONTENT;
+  let responseBodyToDocument = {};
+  await BasicServices.updateClient(body, user, xCorrelator, traceIndicator, customerJourney, req.url, NEW_RELEASE_FORWARDING_NAME)
```
Comment (PrathibaJee) [inline server/controllers/BasicServices.js:269]: undefined variable "**NEW_RELEASE_FORWARDING_NAME**" causing the following error and terminating the application , 

```

ReferenceError: NEW_RELEASE_FORWARDING_NAME is not defined
    at SwaggerRouter.redirectServiceRequestInformation (..\AirInterfacePowerSaver\server\controllers\BasicServices.js:185:124)
    at ..\AirInterfacePowerSaver\server\node_modules\openbackhaul-oas3-tools\dist\middleware\swagger.router.js:99:40
    at Layer.handle [as handle_request] (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\layer.js:95:5)
    at trim_prefix (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\index.js:328:13)
    at ..\AirInterfacePowerSaver\server\node_modules\express\lib\router\index.js:286:9
    at Function.process_params (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\index.js:346:12)
    at next (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\index.js:280:10)
    at ..\AirInterfacePowerSaver\server\node_modules\openbackhaul-oas3-tools\dist\middleware\swagger.parameters.js:52:13
    at Layer.handle [as handle_request] (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\layer.js:95:5)
    at trim_prefix (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\index.js:328:13)
```

File: server/controllers/BasicServices.js [hunk 5]
```diff
@@ -1,164 +1,320 @@
...
 
-module.exports.updateOperationClient = function updateOperationClient (req, res, next, body, user, originator, xCorrelator, traceIndicator, customerJourney) {
-  BasicServices.updateOperationClient(body, user, originator, xCorrelator, traceIndicator, customerJourney)
-    .then(function (response) {
-      utils.writeJson(res, response);
+module.exports.updateOperationClient = async function updateOperationClient (req, res, next, body, user, originator, xCorrelator, traceIndicator, customerJourney) {
+  let startTime = process.hrtime();
+  let responseCode = responseCodeEnum.code.NO_CONTENT;
+  let responseBodyToDocument = {};
+  await BasicServices.updateOperationClient(body, user, xCorrelator, traceIndicator, customerJourney, req.url, NEW_RELEASE_FORWARDING_NAME)
```
Comment (PrathibaJee) [inline server/controllers/BasicServices.js:288]: undefined variable "**NEW_RELEASE_FORWARDING_NAME**" causing the following error and terminating the application , 

```

ReferenceError: NEW_RELEASE_FORWARDING_NAME is not defined
    at SwaggerRouter.redirectServiceRequestInformation (..\AirInterfacePowerSaver\server\controllers\BasicServices.js:185:124)
    at ..\AirInterfacePowerSaver\server\node_modules\openbackhaul-oas3-tools\dist\middleware\swagger.router.js:99:40
    at Layer.handle [as handle_request] (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\layer.js:95:5)
    at trim_prefix (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\index.js:328:13)
    at ..\AirInterfacePowerSaver\server\node_modules\express\lib\router\index.js:286:9
    at Function.process_params (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\index.js:346:12)
    at next (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\index.js:280:10)
    at ..\AirInterfacePowerSaver\server\node_modules\openbackhaul-oas3-tools\dist\middleware\swagger.parameters.js:52:13
    at Layer.handle [as handle_request] (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\layer.js:95:5)
    at trim_prefix (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\index.js:328:13)
```

File: server/controllers/BasicServices.js [hunk 6]
```diff
@@ -1,164 +1,320 @@
...
 
-module.exports.inquireOamRequestApprovals = function inquireOamRequestApprovals (req, res, next, body, user, originator, xCorrelator, traceIndicator, customerJourney) {
-  BasicServices.inquireOamRequestApprovals(body, user, originator, xCorrelator, traceIndicator, customerJourney)
-    .then(function (response) {
-      utils.writeJson(res, response);
+module.exports.inquireOamRequestApprovals = async function inquireOamRequestApprovals (req, res, next, body, user, originator, xCorrelator, traceIndicator, customerJourney) {
+  let startTime = process.hrtime();
+  let responseCode = responseCodeEnum.code.NO_CONTENT;
+  let responseBodyToDocument = {};
+  await BasicServices.inquireOamRequestApprovals(body, user, xCorrelator, traceIndicator, customerJourney, req.url, NEW_RELEASE_FORWARDING_NAME)
```
Comment (PrathibaJee) [inline server/controllers/BasicServices.js:128]: undefined variable "**NEW_RELEASE_FORWARDING_NAME**" causing the following error and terminating the application , 

```

ReferenceError: NEW_RELEASE_FORWARDING_NAME is not defined
    at SwaggerRouter.redirectServiceRequestInformation (..\AirInterfacePowerSaver\server\controllers\BasicServices.js:185:124)
    at ..\AirInterfacePowerSaver\server\node_modules\openbackhaul-oas3-tools\dist\middleware\swagger.router.js:99:40
    at Layer.handle [as handle_request] (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\layer.js:95:5)
    at trim_prefix (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\index.js:328:13)
    at ..\AirInterfacePowerSaver\server\node_modules\express\lib\router\index.js:286:9
    at Function.process_params (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\index.js:346:12)
    at next (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\index.js:280:10)
    at ..\AirInterfacePowerSaver\server\node_modules\openbackhaul-oas3-tools\dist\middleware\swagger.parameters.js:52:13
    at Layer.handle [as handle_request] (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\layer.js:95:5)
    at trim_prefix (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\index.js:328:13)
```

**Review by PrathibaJee — CHANGES_REQUESTED** [review]
Thanks for the changes. Kindly please incorporate the review comments. Also requesting you to perform unit testing on the following services,
1. /v1/redirect-service-request-information
2. /v1/redirect-oam-request-information
3. /v1/inquire-oam-request-approvals
4. /v1/update-client
5. /v1/redirect-topology-change-information
6. /v1/update-operation-client
 as they are failing with 

image

File: server/controllers/BasicServices.js (hunk 1 above)
Comment (at00825957) [reply server/controllers/BasicServices.js:167]: Done!

File: server/controllers/BasicServices.js (hunk 2 above)
Comment (at00825957) [reply server/controllers/BasicServices.js:186]: Done!

File: server/controllers/BasicServices.js (hunk 3 above)
Comment (at00825957) [reply server/controllers/BasicServices.js:205]: Done!

File: server/controllers/BasicServices.js (hunk 4 above)
Comment (at00825957) [reply server/controllers/BasicServices.js:270]: Done!

File: server/controllers/BasicServices.js (hunk 5 above)
Comment (at00825957) [reply server/controllers/BasicServices.js:289]: Done!

File: server/controllers/BasicServices.js (hunk 6 above)
Comment (at00825957) [reply server/controllers/BasicServices.js:129]: Done!

**Review by PrathibaJee — APPROVED** [review]

**Pull request merged**